lib/models/movie.dart

```diff
@@ -39,5 +53,36 @@ class Movie {
         ],
         'images/SuperMarioBackdrop.jpg',
         'images/SuperMarioPoster.jpg',
+        false,
+        'en',
+        0,
+        100,
+        false,
+        100000,
       );
+
+  factory Movie.fromJson(Map<String, dynamic> json, List<Genre> genres) {
+    List<dynamic> movieGenreIDs = json['genre_ids'] as List<dynamic>;
+    List<String> movieGenres = genres
+        .where((e) => movieGenreIDs.contains(e.id))
+        .map((e) => e.name)
+        .toList();
```

Comment:
you could create a factory method fromList to get the genres inside the Genre class, like this

```
static List<Genre>? fromList(List<dynamic>? list) {
    var result = <Genre>[];
    list?.forEach(
      (item) {
        var genre = Genre.fromJson(item);
        if (genre != null) {
          result.add(genre);
        }
      },
    );

    return result;
  }
```

Reply:
Is it best to leave this in the Genre class or use a genresFromList method in Movie? It makes sense for it to be in Genres but when called "Genre.fromList()" sounds like it should return a Genre, not a list of Genres.

Reply:
A second question now that I'm re-reading the code...Would it be better if the Genre class had a list with all the possible Genres, and a static "getGenre(id)" method that I could call from the Movie, or should I keep the current approach of passing a List of possible Genres to the Movie, and having the Movie decide which ones it has?

Reply:
for the first question, its better to have it on Genre, to keep the related methods according to the class, its not wrong that Genre.fromList returns a list of Genres, better for Movie to receive the info that to transform it.
And for the second, the method sounds better, you could use it inside the fromJson

Reply:
Sorted, thank you!